src/bit_array.cr

```diff
@@ -165,6 +165,54 @@ struct BitArray
     end
   end
 
+  # :inherit:
+  def fill(value : Bool) : self
+    return self if size == 0
+
+    if size <= 32
+      @bits.value = value ? ~(UInt32::MAX << size) : 0_u32
+    elsif size <= 64
+      @bits.as(UInt64*).value = value ? ~(UInt64::MAX << size) : 0_u64
+    else
+      to_slice.fill(value ? 0xFF_u8 : 0x00_u8)
+      clear_unused_bits if value
+    end
+
+    self
+  end
+
+  # :inherit:
+  def fill(value : Bool, start : Int, count : Int) : self
+    start, count = normalize_start_and_count(start, count)
+    return self if count <= 0
+    bytes = to_slice
+
+    start_bit_index, start_sub_index = start.divmod(8)
+    end_bit_index, end_sub_index = (start + count - 1).divmod(8)
+
+    if start_bit_index == end_bit_index
+      # same UInt8, don't perform the loop at all
+      mask = uint8_mask(start_sub_index, end_sub_index)
+      value ? (bytes[start_bit_index] |= mask) : (bytes[start_bit_index] &= ~mask)
```

Comment:
This is a conditional statement written as a one-line conditional expression. That's not normally considered good style, I think?